saged/normalize_recount_data.py

```diff
@@ -0,0 +1,178 @@
+"""This script normalizes the data from recount3"""
+
+import argparse
+from typing import Dict
+
+import numpy as np
+import tqdm
+
+
+def parse_gene_lengths(file_path: str) -> Dict[str, int]:
+    """Parses a tsv file containing genes and their length
+
+    Arguments
+    ---------
+    file_path - The path to the file mapping genes to lengths
+
+    Returns
+    -------
+    gene_to_len - A dict mapping ensembl gene ids to their lenght in base pairs
+    """
+    gene_to_len = {}
+    with open(file_path) as in_file:
+        # Throw out header
+        in_file.readline()
+        for line in in_file:
+            line = line.replace('"', '')
+            gene, length = line.strip().split('\t')
+            try:
+                gene_to_len[gene] = int(length)
+            except ValueError:
+                # Some genes have no length, but will be removed in a later step
+                pass
+    return gene_to_len
+
+
+def calculate_tpm(counts: np.ndarray, gene_length_arr: np.ndarray) -> np.ndarray:
+    """"Given an array of counts, calculate the transcripts per kilobase million
```

Comment:
I thought TPM just meant "transcripts per million" without considering length, not "transcripts per kilobase million." Am I wrong?

Reply:
That would make sense given the acronym. It seems to be [used interchangeably](https://www.google.com/search?q=transcripts+per+million+vs+transcripts+per+kilobase+million) on the internet, so I'll add a link to the definition I'm working from in the function comments